### PR TITLE
fix(cron): Log excess memory usage at debug and info levels

### LINKED ISSUE #45189: [Bug]: Used memory grew by more than 10 MB when executing job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Every few cron runs I see the warning `Used memory grew by more than 10 MB when executing job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash` in the log. Running with loglevel 0 does not reveal any more details:

```
{
    "reqId": "ai9MUjylJUXj6YXlKtXv",
    "level": 0,
    "time": "2024-05-05T19:41:34+02:00",
    "remoteAddr": "",
    "user": "--",
    "app": "cron",
    "method": "",
    "url": "--",
    "message": "CLI cron call has selected job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash (id: 5, arguments: null)",
    "userAgent": "--",
    "version": "29.0.0.19",
    "data": {
        "app": "cron"
    }
}
 {
    "reqId": "ai9MUjylJUXj6YXlKtXv",
    "level": 0,
    "time": "2024-05-05T19:41:34+02:00",
    "remoteAddr": "",
    "user": "--",
    "app": "cron",
    "method": "",
    "url": "--",
    "message": "Starting job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash (id: 5, arguments: null)",
    "userAgent": "--",
    "version": "29.0.0.19",
    "data": {
        "app": "cron"
    }
}
 {
    "reqId": "ai9MUjylJUXj6YXlKtXv",
    "level": 0,
    "time": "2024-05-05T19:41:36+02:00",
    "remoteAddr": "",
    "user": "--",
    "app": "cron",
    "method": "",
    "url": "--",
    "message": "Finished job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash (id: 5, arguments: null) in 2 seconds",
    "userAgent": "--",
    "version": "29.0.0.19",
    "data": {
        "app": "cron"
    }
}
 {
    "reqId": "ai9MUjylJUXj6YXlKtXv",
    "level": 2,
    "time": "2024-05-05T19:41:36+02:00",
    "remoteAddr": "",
    "user": "--",
    "app": "cron",
    "method": "",
    "url": "--",
    "message": "Used memory grew by more than 10 MB when executing job OCA\\Files_Trashbin\\BackgroundJob\\ExpireTrash (id: 5, arguments: null): 48.7 MB (before: 33 MB)",
    "userAgent": "--",
    "version": "29.0.0.19",
    "data": {
        "app": "cron"
    }
}
```

### Steps to reproduce

Wait for cron to run

### Expected behavior

No error.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "...",
            "..."
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "...",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "version": "29.0.0.19",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "sendmail",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "log_type": "syslog",
        "logtimezone": "Europe\/Berlin",
        "logfile": "",
        "loglevel": 2,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\TIFF"
        ],
        "maintenance": false,
        "theme": "",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "preview_max_x": 1024,
        "preview_max_y": 1024,
        "preview_max_scale_factor": 1,
        "skeletondirectory": "",
        "updater.release.channel": "stable",
        "default_language": "de",
        "app_install_overwrite": [
            "tasks",
            "polls",
            "previewgenerator",
            "forms",
            "spreed",
            "groupfolders"
        ],
        "trashbin_retention_obligation": "auto, 30",
        "default_phone_region": "DE",
        "allow_local_remote_servers": true,
        "debug": false,
        "defaultapp": "files,dashboard"
    }
}
```

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### LINKED ISSUE #45682: [Bug]: Change Warn on excessive memory consumption in background jobs to info
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description
Log file it getting spammed with these warning. i would recommended this to  info and not warning, 
Used memory grew by more than 10 MB when executing job OC\FilesMetadata\Job\UpdateSingleMetadata (id: 175902, arguments: ["usernae",1042806]): 26.1 MB (before: 10.7 MB)

### Steps to reproduce

N/A

### Expected behavior

Should not see this as a warning more info then a warning 

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

image

### PR DESCRIPTION
**Pull request by joshtrichards**
* Resolves: #45189
* Resolves: https://github.com/nextcloud/server/issues/45682

## Summary

Admins/operators don't need to see this in day-to-day scenarios >= the default log level. Moved the 10MiB (common) to DEBUG level and the 300 MiB to INFO level (rarer but still common enough).

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by szaimen — APPROVED** [review]

File: cron.php [hunk 1]
```diff
@@ -164,10 +164,10 @@
 			$memoryPeakAfter = memory_get_peak_usage();
 
 			if ($memoryAfter - $memoryBefore > 10_000_000) {
-				$logger->warning('Used memory grew by more than 10 MB when executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryAfter). ' (before: ' . Util::humanFileSize($memoryBefore) . ')', ['app' => 'cron']);
+				$logger->debug('Used memory grew by more than 10 MB when executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryAfter). ' (before: ' . Util::humanFileSize($memoryBefore) . ')', ['app' => 'cron']);
 			}
 			if ($memoryPeakAfter > 300_000_000) {
-				$logger->warning('Cron job used more than 300 MB of ram after executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryPeakAfter) . ' (before: ' . Util::humanFileSize($memoryPeakBefore) . ')', ['app' => 'cron']);
+				$logger->info('Cron job used more than 300 MB of ram after executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryPeakAfter) . ' (before: ' . Util::humanFileSize($memoryPeakBefore) . ')', ['app' => 'cron']);
```
Comment (ChristophWurst) [inline cron.php:170]: I think warning is fine here. if this grows further things go haywire.

Comment (ChristophWurst) [reply cron.php:170]: A report like https://github.com/nextcloud/server/issues/45189 is what we want. We want to find out when things run inefficiently. We can't always catch those on development environments so we rely on production reporting.

The only things that's to change is the threshold. 10MB memory used in https://github.com/nextcloud/server/issues/45189 is not alarming. If someone saw the >300MB log I very much would like them to notice and report that.

Comment (come-nc) [reply cron.php:170]: Hum we should only warn if the peak after is higher that the one before then, it makes sense.

Comment (nickvergessen) [inline cron.php:170]:
```suggestion
			if ($memoryPeakAfter > 300_000_000 && $memoryPeakBefore <= 300_000_000) {
				$logger->warning('Cron job used more than 300 MB of ram after executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryPeakAfter) . ' (before: ' . Util::humanFileSize($memoryPeakBefore) . ')', ['app' => 'cron']);
```

File: cron.php [hunk 2]
```diff
@@ -164,10 +164,10 @@
 			$memoryPeakAfter = memory_get_peak_usage();
 
 			if ($memoryAfter - $memoryBefore > 10_000_000) {
-				$logger->warning('Used memory grew by more than 10 MB when executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryAfter). ' (before: ' . Util::humanFileSize($memoryBefore) . ')', ['app' => 'cron']);
+				$logger->debug('Used memory grew by more than 10 MB when executing job ' . $jobDetails . ': ' . Util::humanFileSize($memoryAfter). ' (before: ' . Util::humanFileSize($memoryBefore) . ')', ['app' => 'cron']);
```
Comment (nickvergessen) [inline cron.php:167]: As per below I think it makes sense to log in prod when things are consuming (and not freeing) multiple MBs.
Maybe 10 is too small, but we should definetly log this.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**